app/controllers/shared_lists_controller.rb

```diff
@@ -0,0 +1,35 @@
+class SharedListsController < ApplicationController
+  before_action :require_user_logged_in!
+
+  def create
+    list_id = shared_params[:list_id]
+    email = shared_params[:email]
+    user_list_shared_with = User.find_by(email: email)
+    @shared_list = SharedList.new(
+      list_id: list_id,
+      user_id: user_list_shared_with&.id
+    )
+    if @shared_list.save
+      respond_to do |format|
+        format.html { redirect_to root_path,
+          notice: "#{List.find_by(id: list_id).name} shared with #{email}"
+        }
+        format.json { render json: {
+          message: "#{List.find_by(id: list_id).name} shared with #{email}"
```

Comment:
You don't need to retrieve the `List` object this way.  You already have access to it via `@shared_list.list`